homeassistant/components/unifiprotect/switch.py

```diff
@@ -350,42 +360,90 @@ def __init__(
         self._attr_name = f"{self.device.display_name} {self.entity_description.name}"
         self._switch_type = self.entity_description.key
 
-        if not isinstance(self.device, Camera):
-            return
-
-        if self.entity_description.key == _KEY_PRIVACY_MODE:
-            if self.device.is_privacy_on:
-                self._previous_mic_level = 100
-                self._previous_record_mode = RecordingMode.ALWAYS
-            else:
-                self._previous_mic_level = self.device.mic_volume
-                self._previous_record_mode = self.device.recording_settings.mode
-
     @property
     def is_on(self) -> bool:
         """Return true if device is on."""
         return self.entity_description.get_ufp_value(self.device) is True
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the device on."""
-        if self._switch_type == _KEY_PRIVACY_MODE:
-            assert isinstance(self.device, Camera)
+
+        await self.entity_description.ufp_set(self.device, True)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the device off."""
+
+        await self.entity_description.ufp_set(self.device, False)
+
+
+class ProtectPrivacyModeSwitch(RestoreEntity, ProtectSwitch):
+    """A UniFi Protect Switch."""
+
+    device: Camera
+
+    def __init__(
+        self,
+        data: ProtectData,
+        device: ProtectAdoptableDeviceModel,
+        description: ProtectSwitchEntityDescription,
+    ) -> None:
+        """Initialize an UniFi Protect Switch."""
+        super().__init__(data, device, description)
+
+        if self.device.is_privacy_on:
+            extra_state = self.extra_state_attributes or {}
+            self._previous_mic_level = extra_state.get(ATTR_PREV_MIC, 100)
+            self._previous_record_mode = extra_state.get(
+                ATTR_PREV_RECORD, RecordingMode.ALWAYS
+            )
+        else:
             self._previous_mic_level = self.device.mic_volume
             self._previous_record_mode = self.device.recording_settings.mode
-            await self.device.set_privacy(True, 0, RecordingMode.NEVER)
+
+    @callback
+    def _update_previous_attr(self) -> None:
+        if self.is_on:
+            self._attr_extra_state_attributes = {
+                ATTR_PREV_MIC: self._previous_mic_level,
+                ATTR_PREV_RECORD: self._previous_record_mode,
```

Comment:
This doesn't look like valid extra state attributes for this entity. They, mic level and recording mode, should just be sensor values if needed. If we just need to store them to be able to restore the settings on the device, please implement `extra_restore_state_data` instead.

Reply:
They do not make sense as sensors as there are already sensors for mic level and recording mode. These values are what they get restored to when this entity is turned off. 

Also, there is zero documentation on `extra_restore_state_data` and on what it supposed to do as far as I can tell, the data is not exposed to the user. While the values have limited use to the user, it does show the user the values will be restored and I hiding the data from the user serves no purpose except implement a flow leading to a more confusing experience.

Reply:
These measurements are not device native, right? The device doesn't report *previous* values, but the current values. We are not allowed to create new state like that.

Reply:
Yes. The previous state is expected from a user experience though since when you turn off a "Privacy Mode" you want the entity to return to the previous state it was when it was turned on. 

Privacy Mode turns off recording, disables the microphone and adds a black screen in front of the camera so you cannot see anything. Adding/removing the screen is simple, but there needs to be a to persistently the previous mic/recording settings so they can be restored when it is disabled. 

While none of this is "native" to UniFi Protect, this is one of the most used features and there was a reason it was added. Persisting the values means it now can survive a HA reboot since previously they were only stored in memory.

Reply:
It's ok to persist the values but not expose them as state attributes as it's not device native state.